pkg/helm/helm.go

```diff
@@ -67,13 +69,56 @@ func (h *MetalLBChart) GetObjects(crdConfig *metallbv1beta1.MetalLB) ([]*unstruc
 	if err != nil {
 		return nil, err
 	}
-	for _, obj := range objs {
+	for i, obj := range objs {
 		// Set namespace explicitly into non cluster-scoped resource because helm doesn't
 		// patch namespace into manifests at client.Run.
 		objKind := obj.GetKind()
 		if objKind != "PodSecurityPolicy" {
 			obj.SetNamespace(h.namespace)
 		}
+		// patch affinity and resources parameters explicitly into appropriate obj.
+		// This is needed because helm template doesn't support loading non table
+		// structure values.
+		if objKind == "Deployment" {
```

Comment:
I'd change this to "if isControllerDeployment" to make it future proof in case we add other deployments